### ISSUE TITLE
Homebrew formula

### ISSUE THREAD

**Issue by aoberoi** [nilaoda/N_m3u8DL-RE#324]
For macOS (and Linux) users, it would be much easier to install this program if it were distributed on [Homebrew](https://brew.sh/). Currently, a tidy installation on macOS looks like this:
* Downloading the latest release
* Extracting the archive
* Thinking about where on the `PATH` makes sense to place the binary. Doing research.
* Deciding on a place (often times like `/usr/local/bin`). Maybe having to create that directory. Maybe having to update your `PATH` in the appropriate shell init scripts.
* Moving the binary to that directory. Maybe failing since you need to use `sudo`.
* Trying to run the binary. Maybe failing because you need to `chmod +x` the binary.
* Maybe later realizing you should `chown root:wheel` the binary so it matches the permissions of other binaries in that directory.

For experienced users, this seems straightforward and easy. However, for many users who are unfamiliar with the command line, this is confusing and tricky. You could spend hours Googling for solutions, and hit various roadblocks such as determining if your machine is Intel or Arm, how that changes certain assumptions, etc.

If a Homebrew formula is published, this process can be simplified to a couple commands. I'm curious to know if publishing a formula has been considered and if there are any known reasons for not doing it so far.

I'd be happy to help author the formula, if the leaders here are open to that. It seems that we'd want to integrate it into the Actions so that updates to the formula versions occur with as few manual steps as possible (hopefully zero).

PS. Thanks for this awesome project. I'm a new user, and I found it very comprehensive and useful.

**Comment by garnajee** [comment]
Hey,

I’ve created both a Formula version (which builds the package) and a Cask version (which simply downloads the precompiled binary).

Cask versions are automatically updated to the latest available version, but this feature hasn’t been implemented for the Formula version yet.

This is my first Homebrew tap, so if anyone has suggestions for improvements, I’d be happy to merge a pull request!

=> [https://github.com/garnajee/homebrew-perso/](https://github.com/garnajee/homebrew-perso/)

**TL;DR**:

```
brew install --cask garnajee/perso/nm3u8dl-re
# or 
brew install garnajee/perso/nm3u8dl-re
```

---

**Comment by williamcorney** [comment]
Thanks for this
